src/Native/Windows/gen-buildsys-win.bat

```diff
@@ -8,18 +8,29 @@ for %%x in (%*) do Set /A argC+=1
 if NOT %argC%==3 GOTO :USAGE
 if %1=="/?" GOTO :USAGE
 
-setlocal
+setlocal EnableDelayedExpansion
```

Comment:
Question: What does this do?

Reply:
It delays the evaluation of environment variables. Environment variables are expanded at read time of the batch script. When you delay it then they're calculated at execution time whenever you use `!` to surround the environment variable rather than `%`. This means that if you have a for loop or a conditional block, and you want to append a string to the same environment variable on the same block, or on every iteration, you need to use delayed expansion, so that it takes the value in the execution of the conditionals or the loops. If you don't do that, it will just append an empty string. In this particular scenario, this is needed for this piece:

```
else (
  set __VSString=15 2017

  :: VS2017 generator needs to be specified with the arch as part of its name.
  :: x86 is the default arch selected by cmake.
  if /i "%3" == "x64"     (set __VSString=!__VSString! Win64)
  if /i "%3" == "arm"     (set __VSString=!__VSString! ARM)
)
```

Since the IF hasn't been evaluated yet, we need to wait until it happens in order to append the arch to the real string rather than an empty string.